artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java

```diff
@@ -303,6 +307,22 @@ public void setPrefetchSize(int prefetchSize) {
       }
    }
 
+   @Override
+   public void updateForCanceledRef(MessageReference ref) {
+      long seqId = ref.getMessage().getMessageID();
+      long lastDelSeqId = info.getLastDeliveredSequenceId();
+      ServerMessage coreMessage = ref.getMessage();
+      int redeliveryCounter = coreMessage.getIntProperty(OpenWireMessageConverter.AMQ_MSG_REDELIVER_COUNTER);
+      if (openwireDestination.isTopic()) {
+         redeliveryCounter++;
+         coreMessage.putIntProperty(OpenWireMessageConverter.AMQ_MSG_REDELIVER_COUNTER, redeliveryCounter);
```

Comment:
This is wrong anyways Howard, The same message could be delivered to another Queue (another topic for instance).

DeliveryCount is a property of the MessageReference, not message.